genesyscloud/integration_credential/resource_genesyscloud_integration_credential.go

```diff
@@ -82,16 +83,24 @@ func getAllCredentials(ctx context.Context, clientConfig *platformclientv2.Confi
 func createCredential(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	name := d.Get("name").(string)
 	cred_type := d.Get("credential_type_name").(string)
+	fields := buildCredentialFields(d)
+	_, secretFieldPresent := fields["clientSecret"]
```

Comment:
Checking to see if the secretField is set in the fields attribute.  If it is we are not going to attempt to read it later on from the cache.